strawberryfields/io/xir_io.py

```diff
@@ -325,13 +325,21 @@ def _listr(mixed_iterable: Iterable) -> List:
     Any iterable will be cast to a list, including casting all internal types to native Python
     types (e.g., ``Decimal`` and ``np.floating`` to ``float``); Python strings will be cast to
     lists of strings containing a single character each.
+
+    .. note:
+
+        Strings cannot be passed to the function (despite them being iterables). An error will be
+        raised if a string is passed.
     """
+    if isinstance(mixed_iterable, str):
+        raise TypeError("Cannot pass a string.")
```

Comment:
In case only the error message is displayed:
```suggestion
        raise TypeError("Strings cannot be passed to _listr().")
```